src/Components/OATHeader/OATHeader.tsx

```diff
@@ -54,12 +50,72 @@ const OATHeader = ({ elements, dispatch, state }: OATHeaderProps) => {
         link.parentNode.removeChild(link);
     };
 
-    const handleExportClick = () => {
+    const onExportClick = () => {
         const zip = new JSZip();
         for (const element of elements) {
-            let fileName = element['@id'];
-            fileName = fileName.replace(/;/g, '-').replace(/:/g, '_');
-            zip.file(`${fileName}.json`, JSON.stringify(element));
+            const id = element['@id'];
+            let fileName = null;
+            let directoryPath = null;
+
+            // Check if current elements exists within modelsMetadata array, if so, use the metadata
+            // to determine the file name and directory path
+            const modelMetadata = modelsMetadata.find(
+                (model) => model['@id'] === id
+            );
+            if (modelMetadata) {
+                fileName = modelMetadata.fileName
+                    ? modelMetadata.fileName
+                    : null;
+                directoryPath = modelMetadata.directoryPath
+                    ? modelMetadata.directoryPath
+                    : null;
+            }
+
+            // If fileName or directoryPath are null, generate values from id
+            if (!fileName || !directoryPath) {
+                // Get id path - Get section between last ":" and ";"
+                const idPath = id.substring(
```

Comment:
Let's pull out all the logic for parsing DTMIs and file paths into a utils file.

We should also do some more validation around the indexOf return values.